src/sqlite.h.in

```diff
@@ -32,7 +32,69 @@
 */
 #ifndef SQLITE3_H
 #define SQLITE3_H
-#include <stdarg.h>     /* Needed for the definition of va_list */
+
+#ifdef _FREEBSD_KERNEL
+#include <machine/stdarg.h> /* Use this for kernel defines */
+#include <sys/cdefs.h>
+#include <sys/param.h>
+#include <sys/systm.h>
+#include <sys/conf.h>
+#include <sys/event.h>
+#include <sys/fcntl.h> 
+#include <sys/file.h> 
+#include <sys/filedesc.h> 
+#include <sys/filio.h>
+#include <sys/kernel.h>
+#include <sys/kthread.h>
+#include <sys/libkern.h>
+#include <sys/lock.h>
+#include <sys/mutex.h>
+#include <sys/ttycom.h>
+#include <sys/malloc.h>
+#include <sys/module.h>
+#include <sys/namei.h>
+#include <sys/poll.h>
+#include <sys/proc.h>
+#include <sys/random.h>
+#include <sys/selinfo.h>
+#include <sys/signalvar.h>
+#include <sys/stat.h>
+#include <sys/stdint.h>
+#include <sys/syscallsubr.h>
+#include <sys/sysctl.h>
+#include <sys/sysproto.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <sys/uio.h>
+#include <sys/unistd.h>
+#include <sys/user.h>
+#define SQLITE_OS_KV 0
+#define SQLITE_OS_UNIX 0
+#define SQLITE_OS_WIN 0
+#define SQLITE_OS_OTHER 1
+#define assert(X)
+#define SQLITE_ENABLE_HIDDEN_COLUMNS 0
+#define SQLITE_HAVE_ISNAN 0
+#define HAVE_ISNAN 0
+#define SQLITE_OMIT_AUXILIARY_SAFETY_CHECKS 1
+#define SQLITE_OMIT_LOCALTIME 1
+#define SQLITE_OMIT_FLOATING_POINT 1
+#define SQLITE_OMIT_DATETIME_FUNCS 1
+#define HAVE_GMTIME_R 0
+#define HAVE_STRCHRNUL 0
+#define SQLITE_OS_WIN_THREADS 0
+#define HAVE_MALLOC_H 0
+MALLOC_DECLARE(M_SQLITE);
+#define SQLITE_OMIT_WAL 1
+#define SQLITE_ENABLE_STAT4 1
+#define FILENAME_MAX 255
+#define SQLITE_MAX_PATHLEN 255
+#define SQLITE_DEFAULT_CKPTFULLFSYNC 0
+#define CHAR_BIT 8
+
```

Comment:
This is where you would add the SQLITE_DEBUG and SQLITE_OMIT_EXTENSION defines that you have #ifdeef'd out elsewhere.

Reply:
Yes -- they were initially added there, but I removed them because of several kernel issues that were caused by several #ifdef blocks for SQLITE_DEBUG and SQLITE_OMIT_EXTENSION. They are now not defined at all, and I removed the guards, so for cases where we have #if, it just defaults to 0, which is the intended behavior.